shimmingtoolbox/simulate/numerical_model.py

```diff
@@ -107,30 +106,69 @@ def generate_deltaB0(self, field_type, params):
         Defines the starting volume. Sets the background B0 field to zeros.
 
         Args:
-            field_type: Type of field to be generated. Available implementations are: ``'linear'``.
-            params: List of parameters defining the field for the selected
-                field type. If ``field_type = 'linear'``, then ``params`` are
-                ``[m b]`` where m (Hz/pixel) is the slope and b is the floor
-                field (Hz).
+            field_type (str): Type of field to be generated. Available implementations are: ``'x'``, ``'y'``, ``'z'``.
+            params (list): List of parameters defining the field for the selected
+                           field type. If ``field_type = 'x' or 'y' or 'z'``, then ``params`` are
+                           ``[m b]`` where m (Hz/pixel) is the slope and b is the floor
+                           field (Hz).
         """
 
-        if field_type == "linear":
+        if field_type == "x":
```

Comment:
Would do:
if len(dims) == 2:
        [X, Y] = np.meshgrid(
         np.linspace(-dims[0], dims[0], dims[0]),
         np.linspace(-dims[1], dims[1], dims[1]))
        
if len(dims) == 3:
         [X, Y, Z] = np.meshgrid(
                        np.linspace(-dims[0], dims[0], dims[0]),
                        np.linspace(-dims[1], dims[1], dims[1]),
                        np.linspace(-dims[2], dims[2], dims[2]))

and then check for which gradient to set self.deltaB0 = m * Z + b

Reply:
Good idea